### PR TITLE
doc: Add pipeToSelf in interaction-patterns.md, #27877

### PR DESCRIPTION
**Pull request by patriknw**
Refs #27877

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/interaction-patterns.md [hunk 1]
```diff
@@ -196,6 +196,40 @@ Java
...
+
+When using an API that returns a @scala[`Future`]@java[`CompletionStage`] from an actor it's common that you would
+like to use the value of the in the actor when the @scala[`Future`]@java[`CompletionStage`] is completed. For
+this purpose the `ActorContext` provides a `pipeToSelf` method.
+
+**Example:**
+
+![pipe-to-self.png](./images/pipe-to-self.png)
+
+An actor, `CustomerRepository`, is invoking a method on `CustomerDataAccess` that returns a @scala[`Future`]@java[`CompletionStage`].
```
Comment (chbatey) [inline akka-docs/src/main/paradox/typed/interaction-patterns.md]: Looks good. My only thought is that it is a big chunk of code that could be intimidating for showing one API.

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-docs/src/main/paradox/typed/interaction-patterns.md (hunk 1 above)
Comment (patriknw) [reply akka-docs/src/main/paradox/typed/interaction-patterns.md]: The other interaction pattern examples are equally large.
I see it more as a way to make the patterns more complete by also including this pretty common interaction type with an external Future. Purpose is not so much as reference docs of the `pipeToSelf`.

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**